host/greatfet/peripherals/i2c_bus.py

```diff
@@ -90,3 +94,24 @@ def transmit(self, address, data, receive_length=0):
             data = []
 
         return data
+
+
+    def scan(self):
+        """
+            Sends empty data over the I2C bus, and recieves ACK/NAK
+            in response for valid/invalid addresses.
+        """
+        
+        valid_addresses = []
+        for address in range(128):
+            # Perform the core transfer...
+            self.board.vendor_request_out(vendor_requests.I2C_XFER, value=address>>1,
+                    index=1, data=[])
+            # Read status (ACK/NAK)
+            status = self.board.vendor_request_in(vendor_requests.I2C_GET_STATUS, 
+                    length=1)
+            if status[0] == 0x40:
+                valid_addresses.append(hex(address))
```

Comment:
I suggest returning the addresses as integers.  The caller can convert it to a string with `hex()` later.  Functions in this API that accept an address expect it to be an integer.